gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/MetricContext.java

```diff
@@ -733,7 +733,7 @@ public MetricContext build() {
      */
     public MetricContext buildStrict() throws NameConflictException {
       if(this.parent == null) {
-        this.parent = RootMetricContext.get();
+        hasParent(RootMetricContext.get());
```

Comment:
is this change correct/needed for this PR?

Reply:
Yes. Because we were setting the parent without inheriting its information like tags, which is handled properly by the method `hasParent`